src/shared/is-object.ts

```diff
@@ -0,0 +1,3 @@
+export function isObject(value: unknown): value is object {
+  return value !== null && typeof value === 'object' && !Array.isArray(value);
```

Comment:
A JavaScript array _is_ normally viewed as being an "object". Conceptually it is an object whose keys are all stringified numbers (`[1, 2, 3]` is `{ ["0"]: 1, ["1"]: 2, ["2"]: 3, length: 3 }`).

Reply:
The question would be, when you are checking if something is an object, do you usually want arrays to be taken into consideration for your following logic?

They are effectively objects, but you usually check if something is an array in order to "iterate" over it or use array methods, while you check if something is an object in order to then check if it has certain type shape or property (which will not overlap with the array API in most of the cases).

But I see your point here. I don't have a hard opinion whether the array check should be there or not.

Reply:
> The question would be, when you are checking if something is an object, do you usually want arrays to be taken into consideration for your following logic?

I think this is going to depend on context. I probably would have chosen to push the `!Array.isArray()` check [down to the caller](https://github.com/inspect-js/is-object/issues/1#issuecomment-74918974). Out of interest I looked for the top N npm packages that implement an "is object" check as a function ([is-object](https://www.npmjs.com/package/is-object), [isobject](https://www.npmjs.com/package/isobject)) and they disagreed on this point.

Reply:
Actually, the first one agrees with you https://github.com/inspect-js/is-object/blob/main/index.js.

I will remove that condition for now.